src/main/java/service/Validator.java

```diff
@@ -0,0 +1,21 @@
+package service;
+
+import db.Storage;
+import dto.Transaction;
+import model.OperationEnum;
+
+public class Validator {
+
+    public static void validate(Transaction transaction, Storage storage) throws WrongQuantityException {
+        int productQuantity = storage.getQuantity(transaction.getProduct()).orElse(0);
+
+        if (checkQuantity(transaction, productQuantity)) {
+            throw new WrongQuantityException("Wrong product quantity.");
+        }
+    }
+    private static boolean checkQuantity(Transaction transaction, int productQuantity){
```

Comment:
Optional
```suggestion
    private static boolean isValidQuantity(Transaction transaction, int productQuantity){
```

So the condition will look like
```
if (!isValidQuantity(transaction, productQuantity)) {
```

Reply:
Thanks! Done.